ocs_ci/ocs/cluster.py

```diff
@@ -757,24 +757,39 @@ def get_rebalance_status(self):
                 and states["count"] == total_pg_count
             )
 
-    def wait_for_rebalance(self, timeout=600):
+    def wait_for_rebalance(self, timeout=600, repeat=3):
         """
         Wait for re-balance to complete
 
         Args:
             timeout (int): Time to wait for the completion of re-balance
+            repeat (int): How many times to repeat the check to make sure, it's
+                really completed.
 
         Returns:
-            bool: True if rebalance completed, False otherwise
+            bool: True if re-balance completed, False otherwise
 
         """
         try:
-            for rebalance in TimeoutSampler(
-                timeout=timeout, sleep=10, func=self.get_rebalance_status
-            ):
-                if rebalance:
-                    logger.info("Re-balance is completed")
-                    return True
+            start_time = time.time()
+            for attempt in range(1, repeat + 1):
+                diff_time = int(time.time() - start_time)
+                logger.debug(f"Attempt {attempt} out of {repeat} repeats.")
+                for rebalance in TimeoutSampler(
+                    timeout=timeout - diff_time,
+                    sleep=10,
+                    func=self.get_rebalance_status,
+                ):
+                    if rebalance:
+                        logger.info(
+                            f"Attempt {attempt} for re-balance out of {repeat} is completed"
```

Comment:
@petr-balogh ,This message needs more clarity that we are reattempting rebalance check even though previously we have seen clean status. Currently it sounds as if in the previous attempt cluster was still rebalancing where as that's not the case.

Reply:
I am mentioning that re banalance is completed and the number of the attempts and also out of how many I am going to perform. How you would like to re-phrase this?